app/javascript/mastodon/reducers/filters.js

```diff
@@ -27,6 +32,10 @@ const normalizeFilters = (state, filters) => {
 
 export default function filters(state = ImmutableMap(), action) {
   switch(action.type) {
+  case FILTERS_CREATE_SUCCESS:
+    return normalizeFilter(state, action.filter);
+  case FILTERS_FETCH_SUCCESS:
+    //TODO: handle deleting obsolete filters
```

Comment:
If `normalizeFilters` is supposed to run for both fetch and import, I would move the comment to below the second case, or to the side, so that the cases have no space between them for clarity. Otherwise, you need to `return state` here...

Reply:
`normalizeFilters` imports new filters without removing old ones, which is fine in the `FILTERS_IMPORT` case, but `FILTERS_FETCH_SUCCESS` should replace filters entirely. I forgot about it and will take care of it.